tests/ecmp/test_fgnhg.py

```diff
@@ -141,168 +139,514 @@ def create_fg_ptf_config(ptfhost, ip_to_port, port_list, bank_0_port, bank_1_por
+    for ip in pc_ips:
+        cmd = cmd + " -c '{} {} {}'".format(ipcmd, prefix, ip)
+    for nexthop in ip_to_port.keys():
+        cmd = cmd + " -c 'no {} {} {}'".format(ipcmd, prefix, nexthop)
+    configure_dut(duthost, cmd)
+    time.sleep(3)
+
+    exp_flow_count = {}
+    flows_per_nh = (NUM_FLOWS)/(len(net_ports))
+    for port in net_ports:
+        exp_flow_count[port] = flows_per_nh
+
+    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.{}.fg_to_regular_ecmp.log".format(test_time)
+
+    ptf_runner(ptfhost,
+            "ptftests",
+            "fg_ecmp_test.FgEcmpTest",
+            platform_dir="ptftests",
+            params={"test_case": 'net_port_hashing',
+                "dst_ip": dst_ip,
+                "config_file": FG_ECMP_CFG,
+                "exp_flow_count": exp_flow_count},
+            qlen=1000,
+            log_file=log_file)
+    # Validate that the other 2 prefixes using Fine Grained ECMP were unaffected
+    exp_flow_count = {}
+    flows_per_nh = (NUM_FLOWS)/(len(port_list))
+    for port in port_list:
+        exp_flow_count[port] = flows_per_nh
+    for ip in dst_ip_list:
+        if ip == dst_ip: continue
+        ptf_runner(ptfhost,
+                "ptftests",
+                "fg_ecmp_test.FgEcmpTest",
+                platform_dir="ptftests",
+                params={"test_case": 'initial_hash_check',
+                    "dst_ip": ip,
+                    "exp_flow_count": exp_flow_count,
+                    "config_file": FG_ECMP_CFG},
+                qlen=1000,
+                log_file=log_file)
+
+
+    logger.info("Transition prefix back to fine grained ecmp and validate packets")
+
+    cmd = vtysh_base_cmd
+    for nexthop in ip_to_port.keys():
+        cmd = cmd + " -c '{} {} {}'".format(ipcmd, prefix, nexthop)
+    for ip in pc_ips:
+        cmd = cmd + " -c 'no {} {} {}'".format(ipcmd, prefix, ip)
+    configure_dut(duthost, cmd)
+    time.sleep(3)
+
+    log_file = "/tmp/fg_ecmp_test.FgEcmpTest.{}.regular_to_fg_ecmp.log".format(test_time)
+    ptf_runner(ptfhost,
+            "ptftests",
+            "fg_ecmp_test.FgEcmpTest",
+            platform_dir="ptftests",
+            params={"test_case": 'create_flows',
+                "dst_ip": dst_ip,
+                "exp_flow_count": exp_flow_count,
+                "config_file": FG_ECMP_CFG},
+            qlen=1000,
+            log_file=log_file)
+    # Validate that the other 2 prefixes using Fine Grained ECMP were unaffected
+    for ip in dst_ip_list:
+        if ip == dst_ip: continue
+        ptf_runner(ptfhost,
+                "ptftests",
+                "fg_ecmp_test.FgEcmpTest",
+                platform_dir="ptftests",
+                params={"test_case": 'initial_hash_check',
+                    "dst_ip": ip,
+                    "exp_flow_count": exp_flow_count,
+                    "config_file": FG_ECMP_CFG},
+                qlen=1000,
+                log_file=log_file)
+
+
+def cleanup(duthost, ptfhost):
```

Comment:
cleanup invocation in setup_teardown has to be changed to cleanup(duthost, ptfhost)

Reply:
Can you please provide some more details as to what you are suggesting for modification here?